rageval/metrics/_answer_disambig_f1.py

```diff
@@ -0,0 +1,180 @@
+      year={2023},
+      eprint={2305.06983},
+      archivePrefix={arXiv},
+      primaryClass={cs.CL}
+}
+"""
+
+
+@dataclass
+@add_attribute('mtype', 'AnswerCorrectness')
+@datasets.utils.file_utils.add_start_docstrings(_DESCRIPTION, _KWARGS_DESCRIPTION)
+class AnswerDisambigF1Correctness(Metric):
+    """Estimates the Disambig-F1 between answers and ground truth answers."""
+
+    name = "answer_disambig_f1"
+
+    ALIAS = ['answer_disambig_f1']
+
+    def __init__(self, model: str = "en_core_web_sm"):
+        """
+        Explicitly initialize AnswerDisambigF1Correctness.
+
+        Ensure all parent classes are initialized.
+        Ensure spacy ner model is initialized.
+        """
+        super().__init__()
+        self._required_columns = ['answers', 'gt_answers']
+        self.model = model
+        self.nlp = spacy.load(model)
+
+    def __repr__(self) -> str:
+        """:return: Formatted string representation of the metric."""
+        return f"{self.ALIAS[0]}"
+
+    def _info(self):
+        return datasets.MetricInfo(
+            description=_DESCRIPTION,
+            inputs_description=_KWARGS_DESCRIPTION,
+            citation=_CITATION,
+            homepage="",
+            features=datasets.Features(
+                {
+                    "answers": datasets.Value("string"),
+                    "gt_answers": datasets.Value("string")
+                }
+            ),
+            codebase_urls=["https://github.com/jzbjyb/FLARE"],
+            reference_urls=["https://aclanthology.org/2022.emnlp-main.566"]
+        )
+
+    def _normalize_text(self, s: str) -> str:
+        def remove_articles(text):
+            return re.sub(r'\b(a|an|the)\b', ' ', text)
+
+        def white_space_fix(text):
+            return ' '.join(text.split())
+
+        def remove_punc(text):
+            exclude = set(string.punctuation)
+            return ''.join(ch for ch in text if ch not in exclude)
+
+        def lower(text):
+            return text.lower()
+        return white_space_fix(remove_articles(remove_punc(lower(s))))
+
+    def _ner(self, s: str) -> List[str]:
+        """Extract named entities from the text."""
+        doc = self.nlp(s)
+        ents = doc.ents
+        return [self._normalize_text(e.text) for e in ents]
+
+    def _validate_data(self, dataset: datasets.Dataset) -> bool:
+        """Validate the of the input dataset."""
+        super()._validate_data(dataset)
+        if not all(isinstance(answer, str) for answer in dataset["answers"]):
+            raise ValueError("The type of answers should be a string.")
+        if not all(isinstance(a, List) or not all(isinstance(item, str) for item in a) for a in dataset["gt_answers"]):
+            raise ValueError("The type of gt_answers should be a list of strings.")
+
+    def _f1_score(self, pred: str, ref: str) -> float:
+        """Compute the f1 score between pred and ref."""
```

Comment:
The original version of disambig-F1 in the paper is built on Roberta trained on SQUADV2, and you replace it with exact match?

Reply:
The original metric was presented in [ASQA paper](https://aclanthology.org/2022.emnlp-main.566/), and implemented through [this code](https://github.com/google-research/language/blob/master/language/asqa/scoring.py#L273). And we adopted an [alternative implementation](https://github.com/jzbjyb/FLARE/blob/ec4b06b502b5ab54f3f9236b0112a5d28482e7bb/src/datasets.py#L29) from the paper [Active Retrieval Augmented Generation](https://arxiv.org/abs/2305.06983).